connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java

```diff
@@ -656,48 +586,43 @@ public void testRestartConnectorAndTasksOnlyTasks() throws Exception {
         FutureCallback<ConnectorStateInfo> restartCallback = new FutureCallback<>();
         herder.restartConnectorAndTasks(restartRequest, restartCallback);
         assertEquals(connectorStateInfo, restartCallback.get(1000L, TimeUnit.MILLISECONDS));
-        PowerMock.verifyAll();
+        ArgumentCaptor<TaskStatus> taskStatus = ArgumentCaptor.forClass(TaskStatus.class);
+        verify(statusBackingStore).put(taskStatus.capture());
+        assertEquals(AbstractStatus.State.RESTARTING, taskStatus.getValue().state());
```

Comment:
Can we also assert the task ID here?
``java
assertEquals(taskId, taskStatus.getValue().id());
```

Reply:
Commited and pushed